src/Controllers/MoviesController.cs

```diff
@@ -21,19 +24,37 @@ public IUnitOfWork UnitOfWork
             get { return this._unitOfWork; }
         }
 
-        public IActionResult Index()
+        public IActionResult Index(string movieGenre, string searchString)
         {
+            var genres = _unitOfWork.MovieRepository.GetGenres();
+
             var movies = _unitOfWork.MovieRepository.GetAll();
 
-            var moviesModels = new List<MovieViewModel>();
+            if (!String.IsNullOrEmpty(searchString))
+            {
+                movies = movies.Where(s => s.Title.Contains(searchString));
+            }
+
+            if (!String.IsNullOrEmpty(movieGenre))
+            {
+                movies = movies.Where(x => x.Genre == movieGenre);
+            }
```

Comment:
i think this is not working well. you are filtering first by title and saving the result in a variable called 'movies' and then filtering by genre and saving in that variable too.

Reply:
This way I can filter by title, genre or both. I could put both logics of NullOrEmpty inside a single Where(), but that would not look nice.